addons/product/models/product_attribute.py

```diff
@@ -118,8 +124,8 @@ def _unlink_except_used_on_product(self):
     def action_open_related_products(self):
         return {
             'type': 'ir.actions.act_window',
-            'name': _("Related Products"),
-            'res_model': 'product.template',
+            'name': _("Products"),
+            'res_model': 'product.template.attribute.line',
             'view_mode': 'tree,form',
-            'domain': [('id', 'in', self.product_tmpl_ids.ids)],
+            'domain': [('product_tmpl_id.id', 'in', self.product_tmpl_ids.ids), ('attribute_id.id', '=', self.id)],
```

Comment:
```suggestion
            'domain': [('attribute_id', '=', self.id)],
```
I think only this domain would do the job. No? :thinking: